### ISSUE TITLE
[2.0.2] Default Database Values. not working

### ISSUE THREAD

**Issue by wyseburn** [phalcon/cphalcon#10495]
I use Phalconphp 2.0.2

I read Phalcon 2.0.2 released.

Default Database Values
Now in the case that a column has a ‘default’ value declared in the field of the mapped table, this 'default’ value will be used instead of inserting 'NULL’:
$robots = new Robots();
$robots->save(); // use all `default` values

But it is not working.

//TABLE------------------------------------------------------------------
CREATE TABLE `test` (
  `a` int(10) unsigned NOT NULL DEFAULT '0'
) ENGINE=InnoDB DEFAULT CHARSET=utf8

//MODEL------------------------------------------------------------------
```php
class Test extends \Phalcon\Mvc\Model
{
    /**
     *
     * @var integer
     */
    public $a;

    /**
     * Returns table name mapped in the model.
     *
     * @return string
     */
    public function getSource()
    {
        return 'test';
    }

    /**
     * Allows to query a set of records that match the specified conditions
     *
     * @param mixed $parameters
     * @return Test[]
     */
    public static function find($parameters = null)
    {
        return parent::find($parameters);
    }

    /**
     * Allows to query the first record that match the specified conditions
     *
     * @param mixed $parameters
     * @return Test
     */
    public static function findFirst($parameters = null)
    {
        return parent::findFirst($parameters);
    }

}
```

//Working------------------------------------------------------------------
$cls = new Test();
$cls->save();

result :
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'a' cannot be null

**Comment by andresgutierrez** [comment]
This was fixed in 2.0.3 (current master):

```php
<?php

namespace Some;

use Phalcon\DI,
    Phalcon\Db\Column,
    Phalcon\Db\RawValue,
    Phalcon\Mvc\Model,
    Phalcon\Events\Manager as EventsManager,
    Phalcon\Db\Dialect\MySQL as SqlDialect,
    Phalcon\Db\Adapter\Pdo\MySQL as Connection,
    Phalcon\Mvc\Model\Manager as ModelsManager,
    Phalcon\Mvc\Model\Metadata\Memory as ModelsMetaData;

$eventsManager = new EventsManager();

$di = new DI();

$connection = new Connection(array(
    "host"     => "localhost",
    "username" => "root",
    "password" => "",
    "dbname"   => "phalcon_test",
));

$connection->setEventsManager($eventsManager);

$eventsManager->attach('db',
    function ($event, $connection) {
        switch ($event->getType()) {
            case 'beforeQuery':
                echo $connection->getSqlStatement(), "\n";
                print_r($connection->getSqlVariables());
                break;
        }
    }
);

$modelsManager = new ModelsManager();
$modelsManager->setDi($di);
$di['db'] = $connection;
$di['modelsManager'] = $modelsManager;
$di['modelsMetadata'] = new ModelsMetadata();

class Test extends \Phalcon\Mvc\Model
{
    /**
     *
     * @var integer
     */
    public $a;

    /**
     * Returns table name mapped in the model.
     *
     * @return string
     */
    public function getSource()
    {
        return 'test';
    }

    /**
     * Allows to query a set of records that match the specified conditions
     *
     * @param mixed $parameters
     * @return Test[]
     */
    public static function find($parameters = null)
    {
        return parent::find($parameters);
    }

    /**
     * Allows to query the first record that match the specified conditions
     *
     * @param mixed $parameters
     * @return Test
     */
    public static function findFirst($parameters = null)
    {
        return parent::findFirst($parameters);
    }
}

$cls = new Test();
$cls->save();
```

Outputs:

```php
SELECT IF(COUNT(*) > 0, 1, 0) FROM `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_NAME` = 'test' AND `TABLE_SCHEMA` = DATABASE()
DESCRIBE `test`
INSERT INTO `test` (`a`) VALUES (?)
Array
(
    [0] => 0
)
```

**Comment by wyseburn** [comment]
Thank you for the great advice. I really appreciate it.
